### ISSUE TITLE
Problem on Emerald ROMs.

### ISSUE THREAD

**Issue by Syreldar** [kimwnasptd/OWM-Qt#6]
I can't seem to use the program on Emerald properly.

https://i.imgur.com/mBJcvQ1.png

The screen has been taken from the analysis of a fresh and untouched Emerald ROM.

As you can see it doesn't load the table properly, it only loads up to 59 sprites, although there are like 240+, And if I try to add a new sprite It corrupts the memory of the ROM.

Update: It's confirmed that it is an issue from the software, a friend of mine just sent me his ROM which works perfectly on his PC, but only shows 60 sprites on mine.

Debug output:
```
Opened a new ROM: C:/Users/X/Desktop/Pokemon Emerald.gba
0x8E6D8

root: About to check: 0x8E6D8 (0x505620)
Found OWs: 60 | Not OW Pointer: 0x505710 | Pointing to: 0x50A1E8
Found Addresses: 0x481F70 0xE44214 0xE46618 0xE71574

tbl_init: OW Pointers(WR): 0x481F70
tbl_init: OW Data(WR): 0xE44214
tbl_init: Frames Pointers(WR): 0xE46618
tbl_init: Frames Address(WR): 0xE71574

root: About to check: 0x8E6D8 (0x481F70)

root: Not a ptr: 0x8E6D8 (0x481F70)
Repointing the Palette Table
```

**Comment by kimwnasptd** [comment]
Hey @ExoryxHana, I haven't looked the code for a long time but we can give it a try debugging your issue.

From the `Found OWs: 60 | Not OW Pointer: 0x505710 | Pointing to: 0x50A1E8` statement we see that when OWM checks pointer at `0x505710` it decides that it is not an [OW Data Pointer](https://github.com/kimwnasptd/OWM-Qt/wiki/Inner-Workings#structures-and-slang). This check happens [here](https://github.com/kimwnasptd/OWM-Qt/blob/master/core_files/core.py#L847). So either the first check [is_ptr](https://github.com/kimwnasptd/OWM-Qt/blob/master/core_files/rom_api.py#L105) fails or [is_ow_data](https://github.com/kimwnasptd/OWM-Qt/blob/master/core_files/core.py#L55). Most probably it is the latter.

Can you confirm that address `0x505710` is a Pointer? If it is, what are the first 36 bytes from where the pointer at this address is pointing to (`OW Data`)?

**Comment by Syreldar** [comment]
**0x505710:**

![HxD_2019-06-02_13-42-55](https://user-images.githubusercontent.com/42327659/58760776-0848d180-853d-11e9-8c1c-045c101ffaca.png)

![HxD_2019-06-02_13-47-23](https://user-images.githubusercontent.com/42327659/58760803-2c0c1780-853d-11e9-97e3-066b85b50bfd.png)


**OW Data(WR): 0xE44214:**

![HxD_2019-06-02_13-44-00](https://user-images.githubusercontent.com/42327659/58760778-0d0d8580-853d-11e9-9865-4aae298ecf09.png)

FF bytes only from here onwards.

Tell me if I misunderstood your words and you need something else. Cause I'm a bit sleepy.

**Comment by kimwnasptd** [comment]
Thanks, these are the offsets I meant. It seems like there is no `OW Data` for the OW at `0x505710` and that's why OWM stops there. 

Could you provide some more info about the ROM? What I want to know is:
* Are these values taken after opening the ROM with OWM?
* Have the OWs been edited with another tool prior to opening the ROM with OWM?
    * Was there any change/repointing of the OW 61? (the one with no `OW Data`)
    * Are you able to edit that OW with the other tool? What `OW Data` is it using?

**Comment by Syreldar** [comment]
1. Yes.
2. No.
3. No change, and no repointing either, the ROM is totally fresh. 
4. Yes. About the data well, look:

This is the last sprite OWM shows to me

![NSE_2019-06-02_15-50-49](https://user-images.githubusercontent.com/42327659/58762240-83ff4a00-854e-11e9-9ca9-436c16acc5e1.png)


I noticed that this sprite (which is the immediate next one) has a strange pointer2, maybe that's why OWM stops before it?

![NSE_2019-06-02_15-50-52](https://user-images.githubusercontent.com/42327659/58762248-9a0d0a80-854e-11e9-9003-587577f5775c.png)

**Comment by kimwnasptd** [comment]
Yes, it turns out that this must be the problem. In (some?) Emerald Roms the pointer at the OW Data + 0x14 might be 0. I tried it with the first Emerald Rom I found and loaded 60 sprites to me also. 

Added this case in the function, it managed to read all the OWs in my ROM. @ExoryxHana can you try again with the new source code?
